src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerations/AvoidMultipleEnumerationsHelpers.cs

```diff
@@ -389,7 +410,10 @@ public static ImmutableArray<IMethodSymbol> GetLinqMethods(WellKnownTypeProvider
             {
                 foreach (var methodName in methodNames)
                 {
-                    builder.AddRange(type.GetMembers(methodName).OfType<IMethodSymbol>());
+                    foreach (var methodSymbol in type.GetMembers(methodName).OfType<IMethodSymbol>())
+                    {
+                        builder.Add(methodSymbol);
+                    }
                 }
```

Comment:
I meant to avoid nested loop:
```cs
    foreach (var methodSymbol in type.GetMembers().OfType<IMethodSymbol>())
    {
        if (methodNames.Contains(methodSymbol.Name))
        {
            builder.Add(methodSymbol);
        }
    }
```